clang/lib/Parse/ParseDecl.cpp

```diff
@@ -4503,7 +4535,8 @@ void Parser::ParseStructUnionBody(SourceLocation RecordLoc,
     FieldDecl *FieldDecl = Pair.first;
     std::unique_ptr<CachedTokens> Tokens = std::move(Pair.second);
     BoundsAnnotations Annots;
-    if (DeferredParseBoundsExpression(std::move(Tokens), Annots, DeclaratorsInfo.D))
+    if (DeferredParseBoundsAnnotations(std::move(Tokens),
+                                       Annots, DeclaratorsInfo.D))
```

Comment:
In another place where `DeferredParseBoundsAnnotations` is called, `ActOnInvalidBoundsDecl` and `ActOnBoundsDecl` are called only when the first token in `Tokens` is not `_Where`. Is such a check not required here? If it is not required, maybe a comment here will be helpful?